### PR TITLE
Issue #13345: Enable Test Example For IndentationCheck

### PR DESCRIPTION
**Pull request by Anmol202005**
Title: Enable examples tests for IndentationCheck
Related Issue: https://github.com/checkstyle/checkstyle/issues/13345

Summary
This pull request aims to enable example tests for the ** IndentationCheck ** in the project. The implementation focuses on ensuring that Indentation Checks are correctly validated through comprehensive test cases, improving overall code reliability and quality.

Key Changes Made

Conversion of txt file to Java File
Enable the test cases by adding expected violations in the expected array for all example tests
Made corresponding changes in xml.template file

### CODE REVIEW

**Comment by romani** [comment]
Please read and watch videos at https://checkstyle.org/beginning_development.html#Starting_Development
You should not close PR if something should be changed, videos shows how to do this.

**Comment by romani** [comment]
GitHub, generate website

**Comment by romani** [comment]
@Anmol202005 , thanks a lot, please minimize a diff between examples https://www.diffchecker.com/r1ockxTa/

See issue description for more examples.

**Comment by Anmol202005** [comment]
I have modified the examples

**Comment by romani** [comment]
GitHub, generate website

**Comment by Anmol202005** [comment]
Could you please review the changes to see if they are ready to be merged? Let me know if any adjustments are needed before proceeding.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    <module name="Indentation"/>
+  </module>
+</module>
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation;
+
+// xdoc section -- start
+class Example1 {
+    String field = "example";// basicOffset
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java:13]: Please add space before each `//` and space after to make easy to read.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        boolean cond6 = false;
+
+        if (cond1
+            || cond2) {
+            field = field.toUpperCase()
+                .concat(" TASK");
+        }
+
+        if ((cond1 && cond2)
+                || (cond3 && cond4) //ok,lineWrappingIndentation
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java:40]: Please put space after `,` in comments, to ease reading

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example2.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  </module>
+</module>
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation;
+
+// xdoc section -- start
+class Example2 {
+    String field = "example";// basicOffset
+    int[] values = {// basicOffset
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example2.java:16]: I likes how it was before, when comments are vertically aligned by same column.

**Review by romani — CHANGES_REQUESTED** [review]
Items

**Comment by romani** [comment]
GitHub, generate website

**Comment by romani** [comment]
@Anmol202005, please reply to each review item as "done". It helps a lot, it will be more meaningful in more complicated updates.

**Comment by romani** [comment]
I still do not see items of my review https://github.com/checkstyle/checkstyle/pull/16067#pullrequestreview-2524784330 to be done.

Please self review  code of your PR in web.

**Comment by Anmol202005** [comment]
Sure Romani, I am on it

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        }
+
+        if ((cond1 && cond2)
+                || (cond3 && cond4)                              // ok, lineWrappingIndentation
+                || !(cond5 && cond6)) {                           // ok, lineWrappingIndentation
+            field.toUpperCase()
+                 .concat(" TASK")                                 // ok, lineWrappingIndentation
+                 .chars().forEach(c -> {                          // ok, lineWrappingIndentation
+                     System.out.println((char) c);
+                 });
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java:46]:
```
        if ((cond1 && cond2)
                || (cond3 && cond4)                // ok, lineWrappingIndentation
                || !(cond5 && cond6)) {            // ok, lineWrappingIndentation
            field.toUpperCase()
                 .concat(" TASK")                  // ok, lineWrappingIndentation
                 .chars().forEach(c -> {           // ok, lineWrappingIndentation
                     System.out.println((char) c);
```

please move comments blocks close to code, by blocks

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        20,
+        30
+    };
+
+    void processValues() throws Exception {
+        handleValue("Test String", 42);                              // basicOffset
+    }
+
+    void handleValue(String aFooString,
+                     int aFooInt) {                                 // indent:8 ; expected: > 4; ok, because 8 > 4
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java:25]:
```
class Example1 {
    String field = "example";                // basicOffset
    int[] values = {                         // basicOffset
        10,
        20,
        30
    };

    void processValues() throws Exception {
        handleValue("Test String", 42);      // basicOffset
    }

    void handleValue(String aFooString,
                     int aFooInt) {          // indent:8 ; expected: > 4
```

File: config/checkstyle-examples-suppressions.xml [hunk 6]
```diff
@@ -104,6 +112,10 @@
   <!-- Example needs to exclude fully qualified name longer than 85 characters  -->
   <suppress checks="LineLength" files="classdataabstractioncoupling[\\/]Example11.java"/>
 
+  <!-- Examples need spacing to be more readable -->
+  <suppress checks="LineLength"
+    files="indentation[\\/]Example[1-5].java"/>
```
Comment (romani) [inline config/checkstyle-examples-suppressions.xml:117]: lets avoid this

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java (hunk 5 above)
Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java:25]: I intentionally cut "indent:8 ; expected: > 4".

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java (hunk 1 above)
Comment (Anmol202005) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java]: done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java (hunk 2 above)
Comment (Anmol202005) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java]: done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example2.java (hunk 3 above)
Comment (Anmol202005) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example2.java]: done

File: config/checkstyle-examples-suppressions.xml (hunk 6 above)
Comment (Anmol202005) [reply config/checkstyle-examples-suppressions.xml]: done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java (hunk 5 above)
Comment (Anmol202005) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java]: done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java (hunk 4 above)
Comment (Anmol202005) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java]: done

**Comment by romani** [comment]
GitHub, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  </module>
+</module>
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation;
+
+// xdoc section -- start
+class Example1 {
+    String field = "example";                  // basicOffset
+    int[] values = {                            // basicOffset
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java:14]: Comment should start from same column

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+                .concat(" TASK");
+        }
+
+        if ((cond1 && cond2)
+                || (cond3 && cond4)          // ok, lineWrappingIndentation
+                || !(cond5 && cond6)) {       // ok, lineWrappingIndentation
+            field.toUpperCase()
+                 .concat(" TASK")             // ok, lineWrappingIndentation
+                 .chars().forEach(c -> {       // ok, lineWrappingIndentation
+                     System.out.println((char) c);
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java:45]: Please put // on same column to make them look pretty, vertical alignment

**Review by romani — CHANGES_REQUESTED** [review]
Last

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java (hunk 4 above)
Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java]: Alignment of comments please

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java (hunk 8 above)
Comment (Anmol202005) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java:45]: done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java (hunk 7 above)
Comment (Anmol202005) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java]: done.

**Comment by romani** [comment]
GitHub, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    void handleValue(String aFooString,
+                     int aFooInt) {             // indent:8 ; expected: > 4;
+        boolean cond1 = true;
+        boolean cond2 = false;
+        boolean cond3 = true;
+        boolean cond4 = false;
+        boolean cond5 = true;
+        boolean cond6 = false;
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java:31]: Can we collapse this block of declarations to single statement, no need to define default values.
It should fit in one line.

So we might not need suppression on file size anymore

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java (hunk 9 above)
Comment (Anmol202005) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java:31]: i have removed the supression .

Comment (Anmol202005) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java:31]: there was no need for the supression.

**Comment by Anmol202005** [comment]
GitHub, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java (hunk 4 above)
Comment (Anmol202005) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java]: done

**Comment by romani** [comment]
Please do https://github.com/checkstyle/checkstyle/pull/16067#discussion_r1899229029

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java (hunk 9 above)
Comment (Anmol202005) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java:31]: Should i leave the variables uninitialized??
if i leave them uninitialized it throws error

Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java:31]: Can you try this https://stackoverflow.com/a/6202907/1015848

Comment (Anmol202005) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example1.java]: done

**Review by romani — APPROVED** [review]
You did update for most complicated Check in whole checkstyle.
Thanks a lot for your time!!!

**Comment by romani** [comment]
GitHub, generate website

**Comment by Anmol202005** [comment]
Welcome , Romani

**Pull request merged**